### DISCUSSION TITLE
UI elements not appearing in  AR mode in ios device(chrome/safari)

### DISCUSSION THREAD

**Discussion by likhita-biyyala** [google/model-viewer#2227] [Q&A]
Hi,

I am trying to change the color of the model on button click.
Here i have created three buttons [red,blue,green] when the user click on each button the color of the model will change to the appropriate color.
It is working perfectly fine in android device.
But, when tried in IOS device (safari/chrome) in AR mode UI elements are not appearing

[glitch-code](https://glitch.com/edit/#!/gentle-precious-effect?path=index.html%3A40%3A47)
 How can i resolve this issue?

**Comment by elalish** [comment]
I'm afraid the only way to resolve this is to ask Apple to implement webXR support in Safari (Chrome on iOS is forced to use the same WebKit engine). This interplay of DOM content and AR is only possible through webXR, which is currently limited to Chrome and Chromium-based Android browsers (it's a recent spec).

**Comment by likhita-biyyala** [comment]
It means interplay  of dom content and AR is only possible in  Chrome (Android devices)?
IOS Chrome?
In 8th wall ui(buttons) elements will appear in ios browser (safari) !!!

**Comment by elalish** [comment]
Indeed, for now. However, webXR is now a web standard, so other browsers are implementing it, it's just a matter of how long it takes. Apple is likely to implement it sooner if WebKit if they hear of users like yourself wanting it. iOS Chrome is in the same boat, as it is forced to use WebKit like Safari. 

I believe 8th Wall's web version uses their own SLAM technology in JS rather than the native ARCore/ARKit, which gives them the flexibility to make these products, but at the cost of performance and download size. Model-viewer is designed to be lightweight, but we'll have to wait for further webXR adoption before we have totally consistent AR across devices.
